### PR TITLE
Prepare for deploying and releasing to maven central

### PR DESCRIPTION
**Pull request by oconnormi**
#### What does this PR do?
* Changes `distributionManagement` to point at the [OSSRH](https://central.sonatype.org/pages/ossrh-guide.html) repository
* Adds release profile
  * generates and attaches javadocs
  * generates and attaches source jars
  * signs artifacts

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
 @mdang8
 @brendan-hofmann
 @mcalcote

#### How should this be tested? (List steps with links to updated documentation)
Through CI (will add a temporary commit that forces the release profile activation)

#### Any background context you want to provide?
As part of this process several configurations and credentials were added to jenkins to support deployment and release. Additionally accounts were created on ossrh and granted permission to deploy to our namespace

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -207,7 +207,12 @@ pipeline {
 
                 withMaven(maven: 'Maven 3.5.4', jdk: 'jdk11', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'cx-oss-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
                     sh 'mvn javadoc:aggregate -B -DskipTests -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS'
-                    sh 'mvn deploy -B -DskipTests -DretryFailedDeploymentCount=10 -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS -Dgpg.secretKeyring=$ION_GPG_KEYRING -Dgpg.publicKeyring=$ION_GPG_KEYRING'
+                    script {
+                        if(params.RELEASE == true) {
+                            sh 'mvn deploy -B -DskipTests -DretryFailedDeploymentCount=10 -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS -Dgpg.secretKeyring=$ION_GPG_KEYRING -Dgpg.publicKeyring=$ION_GPG_KEYRING' -Prelease
+                        } else {
+                            sh 'mvn deploy -B -DskipTests -DretryFailedDeploymentCount=10 -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
```
Comment (brendan-hofmann) [inline Jenkinsfile]: Where does this deploy to?

Comment (oconnormi) [reply Jenkinsfile]: ossrh/central

Comment (oconnormi) [reply Jenkinsfile]: Snapshots end up in the sonatype OSS repo and when releases are performed they end up in central

File: pom.xml [hunk 2]
```diff
@@ -88,17 +84,13 @@
 
     <distributionManagement>
         <snapshotRepository>
-            <id>snapshots</id>
-            <url>${snapshots.repository.url}</url>
+            <id>ossrh</id>
+            <url>/projects/connexta/transformation/pom.xml</url>
```
Comment (brendan-hofmann) [inline pom.xml]: Are we using ION for any of these?

**Comment by oconnormi** [comment]
Latest failure is due to no available transformation-api dependencies. Will need to implement these changes on that repo first

File: pom.xml (hunk 2 above)
Comment (oconnormi) [reply pom.xml]: not sure what you mean exactly? If you mean in the repository id then no, since we are using the ossrh repo provided by sonatype

**Pull request merged**